paddle/pserver/LightNetwork.cpp

```diff
@@ -384,19 +384,22 @@ void SocketClient::TcpClient(const std::string &serverAddr, int serverPort) {
 
   /// Now connect to the server
   int retry_second = 0;
-  int error = 0;
   do {
-    error = connect(sockfd, (sockaddr *)&serv_addr, sizeof(serv_addr));
-    if (error == ECONNREFUSED) {
+    if (connect(sockfd, (sockaddr *)&serv_addr, sizeof(serv_addr)) == 0) {
+      break;
+    }
+
+    if (errno == ECONNREFUSED) {
       LOG(WARNING) << "connection refused by pserver, try again!";
       if (retry_second++ >= 7) {
         LOG(FATAL) << "connection refused by pserver, maybe pserver failed!";
       }
       std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
I think the `retry_second` in the code actually means "how many second we have retried", not  a backoff strategy like "how many second to wait for current retry". Maybe change to `retry_count` is more clear.